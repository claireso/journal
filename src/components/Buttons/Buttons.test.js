import { render } from '@testing-library/react'

import { PrimaryButton } from './index'

describe('<PrimaryButton />', () => {
  it('should render loader', () => {
    const { asFragment } = render(<PrimaryButton isLoading>Button</PrimaryButton>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render button', () => {
    const { asFragment } = render(<PrimaryButton>Button</PrimaryButton>)

    expect(asFragment()).toMatchSnapshot()
  })
})
