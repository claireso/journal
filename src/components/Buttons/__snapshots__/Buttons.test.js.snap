// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> should render button 1`] = `
<DocumentFragment>
  .c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: none;
  color: currentColor;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  outline: none;
  padding: 0;
  margin: 0;
}

.c2 {
  border-radius: 0.2rem;
  font-size: var(--font-size-smaller);
  line-height: 1.3333333;
  padding: 1.6rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
}

.c2 > svg {
  margin: 0 -0.4rem 0 0.4rem;
}

.c0 + .c0 {
  margin: 0 0 0 0.8rem;
}

.c3 {
  background: var(--primary-normal);
  color: white;
}

.c3:hover {
  background: var(--primary-lighter);
}

<button
    class="c0 c1 c2 c3"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`<PrimaryButton /> should render loader 1`] = `
<DocumentFragment>
  .c6 {
  margin: 2rem 0;
  -webkit-animation: iAjNNh 800ms ease-in-out infinite alternate;
  animation: iAjNNh 800ms ease-in-out infinite alternate;
}

.c6:after {
  background: var(--primary-normal);
  border-radius: 50%;
  content: '';
  display: block;
  height: 1rem;
  width: 1rem;
  margin: 0 auto;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: none;
  color: currentColor;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--font-family);
  outline: none;
  padding: 0;
  margin: 0;
}

.c2 {
  border-radius: 0.2rem;
  font-size: var(--font-size-smaller);
  line-height: 1.3333333;
  padding: 1.6rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
}

.c2 > svg {
  margin: 0 -0.4rem 0 0.4rem;
}

.c0 + .c0 {
  margin: 0 0 0 0.8rem;
}

.c3 {
  background: var(--primary-normal);
  color: white;
}

.c3:hover {
  background: var(--primary-lighter);
}

.c4 {
  padding: 1.9rem 1.6rem;
  cursor: default;
}

.c4 .c5 {
  margin: 0;
}

.c4 .c5:after {
  background: #fff;
}

<button
    class="c0 c1 c2 c3 c4"
  >
    <span
      class="c5 c6"
    />
  </button>
</DocumentFragment>
`;
